GoogleSignIn/Tests/Unit/GIDAuthorizationFlowProcessorTest.m

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GoogleSignIn/Sources/GIDAuthorizationFlowProcessor/Implementations/GIDAuthorizationFlowProcessor.h"
+
+#import "GoogleSignIn/Sources/GIDSignInInternalOptions.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthorizationResponse+Testing.h"
+
+#import <XCTest/XCTest.h>
+
+#ifdef SWIFT_PACKAGE
+@import AppAuth;
+@import OCMock;
+#else
+#import <AppAuth/AppAuth.h>
+#import <OCMock/OCMock.h>
+#endif
+
+static NSString *const kFakeURL = @"www.fakeURL.com";
+static NSString *const kErrorDomain = @"ERROR_DOMAIN";
+static NSInteger const kErrorCode = 400;
+
+@interface GIDAuthorizationFlowProcessorTest : XCTestCase {
+  GIDAuthorizationFlowProcessor *_authorizationFlowProcessor;
+  id _authorizationServiceMock;
+  id _externalUserAgentSession;
+}
+
+@end
+
+@implementation GIDAuthorizationFlowProcessorTest
+
+- (void)setUp {
+  [super setUp];
+  
+  _authorizationFlowProcessor = [[GIDAuthorizationFlowProcessor alloc] init];
+  _externalUserAgentSession = OCMProtocolMock(@protocol(OIDExternalUserAgentSession));
+  _authorizationServiceMock = OCMClassMock([OIDAuthorizationService class]);
```

Comment:
If you keep these mocks, then I think you should be verifying things on them. I'm not sure you need to mock, or at least I don't think you need to mock `OIDExternalUserAgentSession`.

Reply:
Added OIDFakeExternalUserAgentSession.

Reply:
Worth verifying the interactions you expect on the existing mock?

Reply:
Added more obvious verification.